Users/Docker_Image.rst

```diff
@@ -1,3 +1,44 @@
+What is Docker ?
+================
+
+Docker is a tool designed to make it easier to create, deploy, and run
+applications by using containers. Containers allow a developer to package
+up an application with all of the parts it needs, like libraries and other
+dependencies, and ship it all out as a single package.
+
+With the help of containers, the developer can rest assured that
+the application will run on other machine regardless of any
+customized settings that machine might have that could differ from the
+machine used for writing and testing the code.
+
+Unlike a virtual machine, rather than creating a whole virtual operating
+system, Docker allows applications to use the same kernel as the system
+that they're running on and only requires applications be shipped
+with things not already running on the host computer. This gives a
+significant performance boost and reduces the size of the application.
+
+Whats is a Docker Image and how is it different from a container ?
+------------------------------------------------------------------
+
+Image is an immutable file and a running instance of an image is called a
+container. Images are composed of layers of other images. Images are created
+when we run ``build`` command and conatiners are formed from these images
+when we use ``run`` command. There can be many containers for the
+same image.
+
+  For more imformation about Docker see the
+  `official documentation <https://docs.docker.com/>`__.
+
+Installing Docker
+-----------------
+
+  Docker installation guide for various operating systems can be found
+  `here <https://docs.docker.com/engine/installation/>`__.
```

Comment:
Optional:
I'm not a fan of using `here`-links, rather describe what's linked, like
```
... can be found in the `official Docker installation instructions <...>`__.
```

But like said, optional and up to you ;)